File: frontend/src/lib/api/feedback.ts
import apiClient from './client'

export interface SatisfactionRating {
  id: number
  user_id: number
  rating: number
  comment: string | null
  created_at: string
  updated_at: string
}

export interface Feedback {
  id: number
  user_id: number
  type: 'bug' | 'feature' | 'other'
  title: string
  description: string
  contact: string | null
  status: 'pending' | 'in_progress' | 'resolved' | 'closed'
  created_at: string
  updated_at: string
}

export interface FeatureRequest {
  id: number
  title: string
  description: string
  status: 'planned' | 'in_progress' | 'completed'
  vote_count: number
  has_voted: boolean
  created_at: string
}

export interface VersionChange {
  type: 'feature' | 'improvement' | 'fix'
  description: string
}

export interface Version {
  id: number
  version: string
  release_date: string
  changes: VersionChange[]
  created_at: string
}

interface ApiResponse<T> {
  data: T
  message?: string
}

export const feedbackApi = {
  getSatisfaction: async (): Promise<{ rating: SatisfactionRating | null }> => {
    const response = await apiClient.get<ApiResponse<SatisfactionRating | null>>('/feedback/satisfaction')
    return { rating: response.data.data }
  },

  submitSatisfaction: async (rating: number, comment?: string): Promise<SatisfactionRating> => {
    const response = await apiClient.post<ApiResponse<SatisfactionRating>>('/feedback/satisfaction', {
      rating,
      comment,
    })
    return response.data.data
  },

  getFeedbacks: async (): Promise<{ data: Feedback[] }> => {
    const response = await apiClient.get<ApiResponse<Feedback[]>>('/feedbacks')
    return { data: response.data.data }
  },

  createFeedback: async (data: {
    type: 'bug' | 'feature' | 'other'
    title: string
    description: string
    contact?: string
  }): Promise<Feedback> => {
    const response = await apiClient.post<ApiResponse<Feedback>>('/feedbacks', data)
    return response.data.data
  },

  getFeatures: async (): Promise<{ data: FeatureRequest[] }> => {
    const response = await apiClient.get<ApiResponse<FeatureRequest[]>>('/features')
    return { data: response.data.data }
  },

  voteFeature: async (featureId: number): Promise<void> => {
    await apiClient.post(`/features/${featureId}/vote`)
  },

  unvoteFeature: async (featureId: number): Promise<void> => {
    await apiClient.delete(`/features/${featureId}/vote`)
  },

  getVersions: async (): Promise<{ data: Version[] }> => {
    const response = await apiClient.get<ApiResponse<Version[]>>('/versions')
    return { data: response.data.data }
  },

  getCurrentVersion: async (): Promise<Version> => {
    const response = await apiClient.get<ApiResponse<Version>>('/versions/current')
    return response.data.data
  },
}

File: backend/internal/models/feedback.go
package models

import "time"

type FeedbackType string

const (
	FeedbackTypeBug     FeedbackType = "bug"
	FeedbackTypeFeature FeedbackType = "feature"
	FeedbackTypeOther   FeedbackType = "other"
)

type FeedbackStatus string

const (
	FeedbackStatusPending    FeedbackStatus = "pending"
	FeedbackStatusInProgress FeedbackStatus = "in_progress"
	FeedbackStatusResolved   FeedbackStatus = "resolved"
	FeedbackStatusClosed     FeedbackStatus = "closed"
)

type Feedback struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	UserID      uint           `gorm:"not null;index" json:"user_id"`
	Type        FeedbackType   `gorm:"not null;size:50" json:"type"`
	Title       string         `gorm:"not null;size:255" json:"title"`
	Description string         `gorm:"not null;type:text" json:"description"`
	Contact     string         `gorm:"size:255" json:"contact,omitempty"`
	Status      FeedbackStatus `gorm:"default:pending;size:50;index" json:"status"`
	AdminReply  string         `json:"admin_reply,omitempty"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
}

func (Feedback) TableName() string {
	return "feedbacks"
}
